angular-ngc/defs.bzl

```diff
@@ -268,30 +274,74 @@ def ng_library(name, package_name = None, deps = [], test_deps = [], visibility
 
     if len(test_spec_srcs) > 0:
         _unit_tests(
-            name = "test",
+            name = "tests",
             tests = test_spec_srcs,
             deps = [":_lib"] + test_deps + TEST_DEPS,
             visibility = visibility,
         )
 
 def _unit_tests(name, tests, deps, visibility):
+    generate_test_setup(name = "test_setup")
+    test_srcs = tests + ["test_setup.ts"]
+
     ng_project(
-        name = "_tests",
-        srcs = tests,
+        name = name,
+        srcs = test_srcs,
         deps = deps,
+        visibility = visibility,
+    )
+
+    karma_polyfills_name = "test_polyfills"
+    generate_karma_polyfills(
+        name = karma_polyfills_name,
+    )
+
+    test_entry = [":%s" % karma_polyfills_name]
+
+    ng_project(
+        name = "_tests_entry",
+        srcs = test_entry,
+        deps = TEST_DEPS,
         testonly = 1,
         visibility = ["//visibility:private"],
     )
 
     # Bundle the spec files
-    esbuild(
+    ng_esbuild(
         name = "_test_bundle",
         testonly = 1,
-        entry_points = [spec.replace(".ts", ".js") for spec in tests],
-        deps = [":_tests"],
+        entry_points = [spec.replace(".ts", ".js") for spec in test_srcs],
+        deps = [name],
         output_dir = True,
         splitting = True,
         visibility = ["//visibility:private"],
     )
 
-    # TODO: create '{name}' target to actually run/test the tests
+    esbuild(
```

Comment:
Isn't this the same as `_test_bundle`? We should only have to bundle the setup + tests once...

Reply:
Oh I see, this is the zonejs bootstrap stuff? None of the naming here implies that as all... test entries are the spec files, nothing to do with zonejs or bootstrapping...